src/pipelines/token_classification.rs

```diff
@@ -0,0 +1,518 @@
+}
+
+impl TokenClassificationConfig {
+    /// Instantiate a new token classification configuration of the supplied type.
+    ///
+    /// # Arguments
+    ///
+    /// * `model_type` - `ModelType` indicating the model type to load (must match with the actual data to be loaded!)
+    /// * model - A `(local name, remote url)` tuple `(&str,&str)` pointing to the model to load (e.g.  model.ot)
+    /// * config - A `(local name, remote url)` tuple `(&str,&str)` pointing to the model configuration to load (e.g. config.json)
+    /// * vocab - A `(local name, remote url)` tuple `(&str,&str)` pointing to the tokenizer's vocabulary to load (e.g.  vocab.txt/vocab.json)
+    /// * vocab - An optional `(local name, remote url)` tuple (`Option<&str,str>`) pointing to the tokenizer's merge file to load (e.g.  merges.txt), needed only for Roberta.
+    ///
+    pub fn new(model_type: ModelType, model: (&str,&str), config: (&str,&str), vocab: (&str,&str), merges: Option<(&str,&str)>) -> TokenClassificationConfig {
+        TokenClassificationConfig {
+            model_type: model_type,
+            model_resource: Resource::Remote(RemoteResource::from_pretrained(model)),
+            config_resource: Resource::Remote(RemoteResource::from_pretrained(config)),
+            vocab_resource: Resource::Remote(RemoteResource::from_pretrained(vocab)),
+            merges_resource: merges.map(|t| { Resource::Remote(RemoteResource::from_pretrained(t)) }),
+            device: Device::cuda_if_available(),
+        }
+    }
+}
+
+impl Default for TokenClassificationConfig {
+    /// Provides a default CONLL-2003 NER model (English)
+    fn default() -> TokenClassificationConfig {
+        TokenClassificationConfig {
+            model_type: ModelType::Bert,
+            model_resource: Resource::Remote(RemoteResource::from_pretrained(BertModelResources::BERT_NER)),
+            config_resource: Resource::Remote(RemoteResource::from_pretrained(BertConfigResources::BERT_NER)),
+            vocab_resource: Resource::Remote(RemoteResource::from_pretrained(BertVocabResources::BERT_NER)),
+            merges_resource: None,
+            device: Device::cuda_if_available(),
+        }
+    }
+}
+
+#[derive(Clone,Copy,Serialize,Deserialize)]
+/// # Identifies the type of model
+pub enum ModelType {
+    Bert,
+    DistilBert,
+    Roberta,
+}
+
+/// # Abstraction that holds a model configuration, can be of any of the supported models
+pub enum ConfigOption {
+    /// Bert configuration
+    Bert(BertConfig),
+    /// DistilBert configuration
+    DistilBert(DistilBertConfig),
+}
+
+/// # Abstraction that holds a particular tokenizer, can be of any of the supported models
+pub enum TokenizerOption {
+    /// Bert Tokenizer
+    Bert(BertTokenizer),
+    /// Roberta Tokenizer
+    Roberta(RobertaTokenizer),
+}
+
+/// # Abstraction that holds one particular token sequence classifier model, for any of the supported models
+pub enum TokenClassificationOption {
```

Comment:
Could this be renamed `ModelOption` to be aligned with `ConfigOption` and `TokenizerOption` ? Or is there a risk of confusion with ModelType?

Reply:
I was thinking more along the lines of a possible conflict if we do the same abstraction for things like sentiment analysis (``SequenceClassificationModel``? After all there's way more than just sentiment analysis using that same paradigm). Parts of the code (``ConfigOption`` and ``TokenizerOption``) would be shared by both, but the models would be separate things.

I do concede that the names are a bit long now :) Am open to better suggestions of course.

Reply:
Alright agreed :) How about `TokenClassificationModelOption` while we're at it? I did not understand at first this was referring to the model and was confusing it with `TokenClassificationConfig` at first glance

Reply:
I only called the things ``*Option`` when they are an enum with multiple options abstracting over multiple models.  Lengthening ``TokenClassificationModel`` in that way too would be a bit confusing I think? (not to mention even longer :) )

Reply:
Agreed - let's leave it as it is